content/en/docs/Writing policies/jmespath.md

```diff
@@ -0,0 +1,309 @@
+---
+title: JMESPath 
+description: The JSON query language behind Kyverno.
+weight: 12
+---
+
+{{% pageinfo color="warning" %}}
+This page is currently under construction.
+{{% /pageinfo %}}
+
+[JMESPath](https://jmespath.org/) (pronounced "James path") is a JSON query language created by James Saryerwinnie and is the language that Kyverno supports to perform more complex selections of fields and values and also manipulation thereof by using one or more [filters](https://jmespath.org/specification.html#filter-expressions). If you're familiar with `kubectl` and Kubernetes already, this might ring a bell in that it's similar to [JSONPath](https://github.com/json-path/JsonPath). JMESPath can be used almost anywhere in Kyverno although is an optional component depending on the type and complexity of a Kyverno policy or rule that is being written. While many policies can be written with simple overlay patterns, others require more detailed selection and transformation. The latter is where JMESPath is useful.
+
+While the complete specifications of JMESPath can be read on the official site's [specifications page](https://jmespath.org/specification.html), much of the specifics may not apply to Kubernetes use cases and further can be rather thick reading. This page serves as an easier guide and tutorial on how to learn and harness JMESPath for Kubernetes resources for use in crafting Kyverno policies. It should not be a replacement for the official JMESPath documentation but simply a use case specific guide to augment the already comprehensive literature.
+
+## Getting Set Up
+
+In order to position yourself for success with JMESPath expressions inside Kyverno policies, a few tools are recommended.
+
+1. `jp`, the JMESPath CLI tool [here](https://github.com/jmespath/jp). This program allows you to test out JMESPath expressions live in a command line interface by passing in a JSON document and seeing the results without having to repeatedly test Kyverno policies.
```

Comment:
Should we just recommend kyverno jp here?

Reply:
Good point. I guess the upstream `jp` is subsumed into `kyverno jp` so doesn't make sense to have two different tools.